include/boost/math/differentiation/autodiff.hpp

```diff
@@ -1261,7 +1264,7 @@ template<typename RealType, size_t Order>
 fvar<RealType,Order> ldexp(const fvar<RealType,Order>& cr, int exp)
 {
     // argument to std::exp2 must be casted to RealType, otherwise std::exp2 returns double (always)
-    return cr * std::exp2(static_cast<RealType>(exp));
+    return cr * std::exp2(static_cast<typename fvar<RealType, Order>::root_type>(exp));
```

Comment:
In this case it may be worth removing the `std::` prefix to allow for ADL, in the case of boost::multiprecision, for instance. (There is a problem with boost::multiprecision::cpp_bin_float in boost 1.69 making it incompatible with autodiff, which is fixed in [develop](https://github.com/boostorg/multiprecision) and presumably for 1.70.)

Reply:
Good call. Updated code is committed.